recipes-app/iot2050-module-firmware-update/files/iot2050-module-firmware-update.tmpl

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) Siemens AG, 2024
+#
+# Authors:
+#  Li Hua Qian <huaqian.li@siemens.com>
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import sys
+import argparse
+import textwrap
+
+def main(args):
+    description=textwrap.dedent('''\
+        Script for updating module firmware on IoT2050
+        Examples usage:
+          1. iot2050-module-firmware-update -s 1 -fwa firmwareA.bin
+          2. iot2050-module-firmware-update -s 1 -fwa firmwareA.bin -fwb firmwareB.bin
+         ''')
+    parser = argparse.ArgumentParser(
+        description=description,
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "-s", "--slot",
+        type=int,
+        required=True,
+        help="Slot number"
+    )
+    parser.add_argument(
+        "-fwa", "--firmware-a",
+        help="Path to firmware A file",
+        type=argparse.FileType('rb')
+    )
+    parser.add_argument(
+        "-fwb", "--firmware-b",
+        help="Path to firmware B file",
+        type=argparse.FileType('rb')
+    )
+    args = parser.parse_args(args)
+
+    if not args.firmware_a and not args.firmware_b:
+        print("Error: No firmware file specified")
+        parser.print_help()
+        return 1
+
+    slot = args.slot
+    slot_path = f"/eiofs/controller/slot{slot}"
```

Comment:
You don't have to define a `slot` for the very next sentence.

Reply:
Done